### PR TITLE
Fix media presentation

### PR DESCRIPTION
**Pull request by michaelkirk**
PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/ViewControllers/MediaDetailViewController.m [hunk 1]
```diff
@@ -44,6 +44,9 @@ - (BOOL)canPerformAction:(SEL)action withSender:(nullable id)sender
 
 #pragma mark -
 
+// This approximates the curve of our message bubbles, which makes the animation feel a little smoother.
+const CGFloat MediaDetailViewControllerMediaCornerRadius = 17;
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/MediaDetailViewController.m]: Can you please pull out a constant in `OWSMessageCell` and use it here?
```
        maskedSubview.layer.cornerRadius = 17;
```

File: Signal/src/ViewControllers/MediaDetailViewController.m [hunk 2]
```diff
@@ -725,19 +732,21 @@ - (void)presentFromViewController:(UIViewController *)viewController
...
+                                   [self applyFinalMediaViewConstraints];
 
                                    // 2. Animate imageView from it's initial position, which should match where it was
                                    // in the presenting view to it's final position, front and center in this view. This
-                                   // animation intentionally overlaps the previous
-                                   [UIView animateWithDuration:0.2
-                                       delay:0.08
+                                   // animation duration intentionally overlaps the previous
+                                   [UIView animateWithDuration:2.2
+                                       delay:2.08
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/MediaDetailViewController.m]: Was this just slowed down for debugging?  2 seconds delay/duration seems very long?

**Comment by charlesmchen** [comment]
It'd be nice to have screenshots or an animated GIF, but NBD.

File: Signal/src/ViewControllers/MediaDetailViewController.m (hunk 2 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/MediaDetailViewController.m]: Yikes, good catch. Thanks.

File: Signal/src/ViewControllers/MediaDetailViewController.m (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/MediaDetailViewController.m]: Yes, done.

**Comment by michaelkirk** [comment]
@charlesmchen I've significantly reworked this to address a couple issues - primarily in 
https://github.com/WhisperSystems/Signal-iOS/pull/2919/commits/3d99f1fe4c443a357782206946c7a2579aa7f17b PTAL

File: Signal/src/ViewControllers/MediaDetailViewController.m [hunk 3]
```diff
@@ -277,7 +280,15 @@ - (void)createContents
     self.mediaView.layer.minificationFilter = kCAFilterTrilinear;
     self.mediaView.layer.magnificationFilter = kCAFilterTrilinear;
 
-    [self applyInitialImageViewConstraints];
+    // The presentationView is only used during present/dismiss animations.
+    // It's a static image of the media content.
+    UIImageView *presentationView = [[UIImageView alloc] initWithImage:self.image];
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/MediaDetailViewController.m]: But isn't `self.image` only true for images?  What about videos?

Comment (charlesmchen) [reply Signal/src/ViewControllers/MediaDetailViewController.m]: gotcha, `image` is a method not a property.

**Pull request merged**